Hypothesis Test For 1 Population Mean

A Z-test is a hypothesis test that compares the mean of your sample data to a known value.

We ca use Z-test if:

+ The population standard deviation ($\sigma$) is known

+ The sample size (n) is larger than 30 (otherwise use a t-test)

+ Your data should be normally distributed. However, for large sample sizes (over 30) this doesn’t always matter

+ Your data should be randomly selected from a population, where each item has an equal chance of being selected.

+ Sample sizes should be equal if at all possible.

Exercise 1: In an anthropological study of animial craniums it was found that the mean circumference of a buffalo's skull is 50 inches with a variance of 25. A pit study at an Indian reservation yielded 64 animals skulls with a mean circumference of 64 inches. Test whether the animal at Indian reservation were buffaloes or not?

Step 1: State the null and the alternative hypothesis

n = 64 

$\bar{X} = 64$

$\sigma^2 = 25$ => $\sigma = 5$

$\mu = 50 inches$

The information does not mention about $\alpha$, we set it as 5% as default

$\alpha = 0.05 $

If the mean circumferences of 64 animals at Indian equals 64 inches, we conclude that they are buffaloes and vice versa

$$H_o: \mu = 50  inches$$ 
$$H_1: \mu ≠ 50  inches$$


Step 2: Compute test statistic $Z_{score}$ and critical value $Z_{crit}$

Since this is a 2 tailed test, we will have 2 critical values ($Z_{crit1}$ and $Z_{crit2}$)

$$Z_{score} = \frac{\bar{X} - \mu}{\frac{\sigma}{\sqrt{n}}}$$

In [36]:
X_bar = 64
mu = 50
sigma = 5
n = 64
alpha = 0.05

# Compute tes statistic Z_score
import math
Z_score = (X_bar-mu)/(sigma/math.sqrt(n))
print("Tes statistic Z_score = ", Z_score)

# Compute critical value
from scipy.stats import norm
Z_crit1 = norm.ppf(alpha/2)
print("Critical value Z_crit1 = ",Z_crit1)

Z_crit2 = norm.ppf(1-alpha/2)
print("Critical value Z_crit2 = ",Z_crit2)



Tes statistic Z_score =  22.4
Critical value Z_crit1 =  -1.9599639845400545
Critical value Z_crit2 =  1.959963984540054


Step 3: Decision making and conclusion

If $Z_{score} \in [Z_{crit1}:Z_{crit2}]$, we do not reject $H_o$

If $Z_{score} > Z_{crit2} $ or $Z_{score} < Z_{crit1} $, we reject $H_o$

Since our $Z_{score} > H_o$ $(22.4 > 1.96)$, we reject $H_o$

Conclusion: The animal at Indian reservation are not buffaloes

We can also compute p-Value to test our data


In [42]:
X_bar = 64
mu = 50
sigma = 5
n = 64
alpha = 0.05

# Compute tes statistic Z_score
import math
Z_score = (X_bar-mu)/(sigma/math.sqrt(n))


If the test is left tailed or right tailed test, we compare p-value with $\alpha$

If the test is two-tailed test, we compare p-value with $\alpha$/2

p-value = 1 - P(Z < $Z_{score}$) when $Z_{score} > 0$

p-value = P(Z < $Z_{score}$) when $Z_{score} < 0$



In [58]:
# Since this is a two-tailed test, we compute p-value = P(Z < Z_score)

# Z (standard normal) distribution has mean = 0 and standard deviation = 1 

p_Value = norm(loc = 0 , scale = 1).cdf(Z_score)
print("p_value = ", p_Value)


p_value =  1.0



We compare p-Value with $\alpha$ when the test is 1 tailed test (right or left tailed)

We compare p-Value with $\frac{\alpha}{2}$ when the test is 2 tailed test 

When p-Value $<$ $\alpha$ or $\frac{\alpha}{2}$, we reject $H_o$

Our p_value = 0.0 (nearly 0 not exact 0), we reject $H_o$

Conclusion: The animal at Indian reservation are not buffaloes

Another test to compare population mean with a known value is t-test

We ca use t-test if:

+ The population standard deviation ($\sigma$) is UNKNOWN or (sample standard deviation sd is known)

+ The sample size (n) is smaller than 30

+ Data is independent

+ Data is collected randomly

+ The data is approximately normally distributed

Exercise 2: To assess the accuracy of a laboratory scale, a standard weight that is known to weigh 1 gram is repeatedly weighed 4 times. The resulting measurements (in grams) are: 0.95, 1.02, 1.01, 0.98. Assume that the weighings by the scale when the true weight is 1 gram are normally distributed with mean µ. Do these data give evidence at 5% significance level that the scale is not accurate? Answer this question by performing an appropriate test of hypothesis.

Step 1: State the two hypothesises

Sample size n = 4

$\mu$ = 1 gram

$\alpha$ = 5%

Since the sample size is smaller than 30 and the population standard deviation $\sigma$ is not provided

We use t-test to test the data

$$H_o: \mu = 1$$
$$H_o: \mu != 1$$

In [59]:
# Create a list containing the sample's results
X = [0.95, 1.02, 1.01, 0.98]

# Compute sample average X_bar and sample standard deviation sd
import statistics

X_bar = statistics.mean(X)
print("Sample average X_bar =", X_bar)

sd = statistics.stdev(X)
print("Sample standard deviation sd =",sd)

Sample average X_bar = 0.99
Sample standard deviation sd = 0.03162277660168382


In [60]:
# Sample size n = 4
n = 4

# Degrees of freedom df = n - 1
df = n - 1

# Significance level alpha = 5%
alpha = 0.05

# Population mean = 1
mean = 1

Step 2: Compute test statistic $t_{score}$ and critical value $t_{crit}$

Since this is a 2 tailed test, we will have 2 critical values ($t_{crit1}$ and $t_{crit2}$)

$$t_{score} = \frac{\bar{X} - \mu}{\frac{sd}{\sqrt{n}}}$$

In [61]:
import math
# Compute test statistic t_score
t_score = (X_bar - mean)/(sd/math.sqrt(n))
print("Test statistic t_score =",t_score)

# Compute the two critical values t_crit1 and t_crit2
from scipy.stats import t

t_crit1 = t.ppf(alpha/2, df)
print("t_crit1 =",t_crit1)

t_crit2 = t.ppf(1 - alpha/2, df)
print("t_crit2 =",t_crit2)

Test statistic t_score = -0.6324555320336759
t_crit1 = -3.1824463052842638
t_crit2 = 3.182446305284263


Step 3: Decision making and conclusion

Since $t_{score}$ $\in$ [$t_{crit1}$ : $t_{crit1}$] 

We do not reject $H_o$

Conclusion: The scale is accurate 

Now, we test data by using p-value

In [62]:
X = [0.95, 1.02, 1.01, 0.98]
mean = 1
n = 4
alpha = 0.05

# Compute sample average X_bar and sample standard deviation sd
import statistics

X_bar = statistics.mean(X)
print("Sample average X_bar =", X_bar)

sd = statistics.stdev(X)
print("Sample standard deviation sd =",sd)

# Compute tes statistic Z_score
import math
t_score = (X_bar-mean)/(sd/math.sqrt(n))
print("Test statistic t_score =",t_score)


Sample average X_bar = 0.99
Sample standard deviation sd = 0.03162277660168382
Test statistic t_score = -0.6324555320336759


If the test is left tailed or right tailed test, we compare p-value with $\alpha$

If the test is two-tailed test, we compare p-value with $\alpha$/2

p-value = 1 - P(t < $t_{score}$) when $t_{score} > 0$

p-value = P(t < $t_{score}$) when $t_{score} < 0$



In [71]:
# Since this is a two-tailed test and t_score < 0,
# we compute p-value = P(t < t_score) and compare it with alpha/2

from scipy.stats import t

# Compute p-value
p_Value = t.cdf(t_score,df)
print("p_value = ",p_Value)


p_value =  0.2860016903503205
